wire-library/wire-runtime-swift/src/test/swift/CodableTests.swift

```diff
@@ -333,6 +333,53 @@ extension CodableTests {
         try assertDecode(json: json, expected: proto)
         try assertEncode(proto: proto, expected: json)
     }
+
+    func testNegativeDurationRoundtrip() throws {
+        let json = """
+        {"duration":"-1.900s"}
+        """
+
+        let proto = DurationAndTimestamp(
+            duration: Wire.Duration(seconds: -2, nanos: 100000000)
+        )
+
+        try assertDecode(json: json, expected: proto)
+        try assertEncode(proto: proto, expected: json)
+    }
+
+    func testLargeDurationRoundtrip() throws {
+        let json = """
+        {
+          "duration":"18014398509481984s",
+          "timestamp":"4001-01-01T00:00:00Z"
+        }
+        """
+        let proto = DurationAndTimestamp(
+            duration: Duration(seconds: 1 << 54, nanos: 0),
+            timestamp: Timestamp(date: .distantFuture)
+        )
+
+        // In theory, we should use a much further distant future; however, there woudl be rounding errors at play
```

Comment:
typo: `woudl`

Also, maybe phrase it as something like `Swift's Date has less precision than Timestamp, so ideally we'd support something further in the future than Date.distantFuture, but our current implementation makes use of Date, and so that's as far as we currently support without encountering rounding errors."